waiter/src/waiter/state.clj

```diff
@@ -753,6 +754,54 @@
   (log/warn "[close-update-state-channel] closing update-state-chan of" service-id)
   (async/go (async/close! update-state-chan)))
 
+(defrecord InstanceTracker [service-id known-instance-ids scheduling-instance-timers starting-instance-trackers])
+
+(defn- new-tracker-val
+  [service-id]
+  (map->InstanceTracker {:service-id service-id
+                         :known-instance-ids #{}
+                         ; TODO - figure out min instances from service-id
+                         ; TODO - track overall request-to-running time
+                         :scheduling-instance-timers [[(timers/start (metrics/waiter-timer "app-schedule-time"))
```

Comment:
Can we name them `service-schedule-time` instead of `app-scheduler-time`?

Reply:
The metric names is something I specifically wanted feedback on. It looks like the metrics can be grouped hierarchically based on the name. Is there a way to name these so they get grouped together?

Reply:
One is a waiter metric while the other is a service metric, they won't be grouped together.

Reply:
I meant that I wanted the service metrics grouped together, and the waiter metrics grouped together. Something like "metrics/service-init-overhead/{startup,schedule,total}".

Reply:
Yeah, you can. `(metrics/waiter-timer "service-init-overhead" "startup")` and `(metrics/waiter-timer "service-init-overhead" "total")`

Reply:
Done.